### PR TITLE
[REVIEW] Pin max `dask` & `distributed` versions

### PR DESCRIPTION
**Pull request by galipremsagar**
This PR introduces max version pinnings for dask & distributed to 2021.11.2

### CODE REVIEW

**Review by bdice — APPROVED** [review]

File: conda/recipes/dask-cudf/meta.yaml [hunk 1]
```diff
@@ -26,13 +26,13 @@ requirements:
...
-    - distributed>=2021.09.1
+    - dask>=2021.09.1,<=2021.11.2
+    - distributed>=2021.09.1,<=2021.11.2
     - cudatoolkit {{ cuda_version }}
   run:
     - python
     - cudf {{ version }}
-    - dask>=2021.09.1
+    - dask>=2021.09.1,<=2021.11.2
     - distributed>=2021.09.1
```
Comment (bdice) [inline conda/recipes/dask-cudf/meta.yaml:36]: @galipremsagar Is this supposed to be pinned as well?
```suggestion
    - distributed>=2021.09.1,<=2021.11.2
```

**Review by bdice — CHANGES_REQUESTED** [review]

File: conda/recipes/dask-cudf/meta.yaml (hunk 1 above)
Comment (galipremsagar) [reply conda/recipes/dask-cudf/meta.yaml:36]: yes, thanks for catching this Bradley!

Comment (galipremsagar) [inline conda/recipes/dask-cudf/meta.yaml]:
```suggestion
    - distributed>=2021.09.1,<=2021.11.2
```

**Review by bdice — APPROVED** [review]

**Review by ajschmidt8 — APPROVED** [review]

File: conda/environments/cudf_dev_cuda11.0.yml [hunk 2]
```diff
@@ -41,8 +41,8 @@ dependencies:
   - pydocstyle=6.1.1
   - typing_extensions
   - pre-commit
-  - dask>=2021.09.1
-  - distributed>=2021.09.1
+  - dask>=2021.09.1,<=2021.11.2
+  - distributed>=2021.09.1,<=2021.11.2
```
Comment (jakirkham) [inline conda/environments/cudf_dev_cuda11.0.yml:45]:
```suggestion
  - dask>=2021.11.1,<=2021.11.2
  - distributed>=2021.11.1,<=2021.11.2
```

File: conda/environments/cudf_dev_cuda11.2.yml [hunk 3]
```diff
@@ -41,8 +41,8 @@ dependencies:
   - pydocstyle=6.1.1
   - typing_extensions
   - pre-commit
-  - dask>=2021.09.1
-  - distributed>=2021.09.1
+  - dask>=2021.09.1,<=2021.11.2
+  - distributed>=2021.09.1,<=2021.11.2
```
Comment (jakirkham) [inline conda/environments/cudf_dev_cuda11.2.yml:45]:
```suggestion
  - dask>=2021.11.1,<=2021.11.2
  - distributed>=2021.11.1,<=2021.11.2
```

File: conda/environments/cudf_dev_cuda11.5.yml [hunk 4]
```diff
@@ -41,8 +41,8 @@ dependencies:
   - pydocstyle=6.1.1
   - typing_extensions
   - pre-commit
-  - dask>=2021.09.1
-  - distributed>=2021.09.1
+  - dask>=2021.09.1,<=2021.11.2
+  - distributed>=2021.09.1,<=2021.11.2
```
Comment (jakirkham) [inline conda/environments/cudf_dev_cuda11.5.yml:45]:
```suggestion
  - dask>=2021.11.1,<=2021.11.2
  - distributed>=2021.11.1,<=2021.11.2
```

File: conda/recipes/custreamz/meta.yaml [hunk 5]
```diff
@@ -31,8 +31,8 @@ requirements:
     - python
     - streamz 
     - cudf {{ version }}
-    - dask>=2021.09.1
-    - distributed>=2021.09.1
+    - dask>=2021.09.1,<=2021.11.2
+    - distributed>=2021.09.1,<=2021.11.2
```
Comment (jakirkham) [inline conda/recipes/custreamz/meta.yaml:35]:
```suggestion
    - dask>=2021.11.1,<=2021.11.2
    - distributed>=2021.11.1,<=2021.11.2
```

File: conda/recipes/dask-cudf/meta.yaml [hunk 6]
```diff
@@ -27,14 +27,14 @@ requirements:
   host:
     - python
     - cudf {{ version }}
-    - dask>=2021.09.1
-    - distributed>=2021.09.1
+    - dask>=2021.09.1,<=2021.11.2
+    - distributed>=2021.09.1,<=2021.11.2
```
Comment (jakirkham) [inline conda/recipes/dask-cudf/meta.yaml:31]:
```suggestion
    - dask>=2021.11.1,<=2021.11.2
    - distributed>=2021.11.1,<=2021.11.2
```

File: conda/recipes/dask-cudf/meta.yaml [hunk 7]
```diff
@@ -27,14 +27,14 @@ requirements:
...
     - cudatoolkit {{ cuda_version }}
   run:
     - python
     - cudf {{ version }}
-    - dask>=2021.09.1
-    - distributed>=2021.09.1
+    - dask>=2021.09.1,<=2021.11.2
+    - distributed>=2021.09.1,<=2021.11.2
```
Comment (jakirkham) [inline conda/recipes/dask-cudf/meta.yaml:37]:
```suggestion
    - dask>=2021.11.1,<=2021.11.2
    - distributed>=2021.11.1,<=2021.11.2
```

File: python/custreamz/dev_requirements.txt [hunk 8]
```diff
@@ -3,8 +3,8 @@
 flake8==3.8.3
 black==19.10b0
 isort==5.6.4
-dask>=2021.09.1
-distributed>=2021.09.1
+dask>=2021.09.1,<=2021.11.2
+distributed>=2021.09.1,<=2021.11.2
```
Comment (jakirkham) [inline python/custreamz/dev_requirements.txt:7]:
```suggestion
dask>=2021.11.1,<=2021.11.2
distributed>=2021.11.1,<=2021.11.2
```

File: python/dask_cudf/dev_requirements.txt [hunk 9]
```diff
@@ -1,7 +1,7 @@
 # Copyright (c) 2021, NVIDIA CORPORATION.
 
-dask>=2021.09.1
-distributed>=2021.09.1
+dask>=2021.09.1,<=2021.11.2
+distributed>=2021.09.1,<=2021.11.2
```
Comment (jakirkham) [inline python/dask_cudf/dev_requirements.txt:4]:
```suggestion
dask>=2021.11.1,<=2021.11.2
distributed>=2021.11.1,<=2021.11.2
```

File: python/dask_cudf/setup.py [hunk 10]
```diff
@@ -10,8 +10,8 @@
 
 install_requires = [
     "cudf",
-    "dask>=2021.09.1",
-    "distributed>=2021.09.1",
+    "dask>=2021.09.1,<=2021.11.2",
+    "distributed>=2021.09.1,<=2021.11.2",
```
Comment (jakirkham) [inline python/dask_cudf/setup.py:14]:
```suggestion
    "dask>=2021.11.1,<=2021.11.2",
    "distributed>=2021.11.1,<=2021.11.2",
```

**Review by jakirkham — COMMENTED** [review]
Per discussion offline suggesting we constrain `dask` & `distributed` with `>=2021.11.1,<=2021.11.2`

**Review by pentschev — APPROVED** [review]
LGTM, thanks @galipremsagar !

**Review by ajschmidt8 — APPROVED** [review]

**Pull request merged**